sunpy/util/tests/test_config.py

```diff
@@ -80,3 +84,68 @@ def test_get_and_create_sample_dir():
     # Set the config back
     os.rmdir(new_path)
     config.set('downloads', 'sample_dir', os.path.join(USER, 'sunpy', 'data', 'sample_data'))
+
+
+def test_write_config_create_new_config(tmpdir, undo_config_dir_patch):
+    config_filename = 'sunpyrc'
+
+    # Try to set a manual one (already created)
+    tmp_config_dir = tmpdir.mkdir("sunpy_test_configdir")
+    os.environ["SUNPY_CONFIGDIR"] = tmp_config_dir.strpath
```

Comment:
Is it possible to do this environment setting in a context manager?

Reply:
Or we can monkeypatch it. Not sure I want to fix it now tho but it would be much better to do so.

Reply:
Done.